sgt-back-end/index.js

```diff
@@ -0,0 +1,126 @@
+const express = require('express');
+const app = express();
+const pg = require('pg');
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+app.get('/api/grades', (req, res, next) => {
+  const grades = [];
+  const sql = `
+    select *
+      from "grades"
+  `;
+  grades.push(sql);
+  db.query(sql)
+    .then(result => {
+      const grades = result.rows;
+      res.status(200).json({ grades });
+    })
+    .catch(error => {
+      console.error(error);
+      res.status(500).json({ error: 'An unexpected error occurrred' });
+    });
+});
+
+const jsonMiddleware = express.json();
+app.use('/api/grades', jsonMiddleware);
+
+app.post('/api/grades', (req, res, next) => {
+  const newGrade = req.body;
+  const grade = Number(newGrade.score);
+  if (!Number.isInteger(grade) || grade < 0 || grade > 100) {
+    res.status(400).json({ error: 'score must be valid (0-100)' });
+    return;
+  } else if (newGrade.name === undefined || newGrade.course === undefined || newGrade.score === undefined) {
+    res.status(400).json({ error: 'Must be a valid input' });
+    return;
+  }
+  const values = [newGrade.name, newGrade.course, newGrade.score];
+  const sql = `
+    insert into "grades" ("name", "course", "score")
+    values ($1, $2, $3)
+    returning *
+  `;
+  db.query(sql, values)
+    .then(result => {
+      res.status(201).json(result.rows);
```

Comment:
The client should receive just the inserted record (object) instead of an array.